var/spack/repos/builtin/packages/mivisionx/package.py

```diff
@@ -34,26 +35,34 @@ def url_for_version(self, version):
 
     variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
 
+    def patch(self):
+        if '@4.2.0' in self.spec:
+            filter_file('${ROCM_PATH}/opencl',
+                        self.spec['rocm-opencl'].prefix,
+                        'amd_openvx/cmake/FindOpenCL.cmake',
+                        string=True)
+
```

Comment:
```suggestion
```

Just set `HIP_PATH` in cmake args, don't add a patch when it's not necessary.

Reply:
done that but it doesn't seem to resolve issue still get the below error:
`     37    -- FindAMDRPP failed to find: amd_rpp
     38    -- WARNING:AMDRPP Not Found -- amd_rpp module excluded
  >> 39    CMake Error at /tmp/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cmake-3.20.2-ar6c2txn2qcsed474wfcnyruxruc2lw5/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
     40      Could NOT find OpenCL (missing: CL_TARGET_OPENCL_VERSION)
     41    Call Stack (most recent call first):
     42      /tmp/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cmake-3.20.2-ar6c2txn2qcsed474wfcnyruxruc2lw5/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
     43      amd_openvx/cmake/FindOpenCL.cmake:28 (find_package_handle_standard_args)
     44      amd_openvx_extensions/amd_loomsl/CMakeLists.txt:26 (find_package)`

I saw the value being set explicitly in the below file:
https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/rocm-4.2.x/amd_openvx/cmake/FindOpenCL.cmake#L99#L108

Reply:
Sigh, that cmake file is just broken. Pinging @kiritigowda.

I mean, the patch works, but the entire custom FindOpenCL.cmake is redundant, as it ONLY works when you hit that branch where it looks for a rocm version of OpenCL in the ROCM_PATH directory. That prefix is not what we have in Spack...

Another solution is to just delete the FindOpenCL.cmake file.

Reply:
I'm going to hold off on merging until we get at least some confirmation that the mivisionx cmake can be fixed in a next release instead of having to add even more patched than we already have in the rocm ecosystem...

Reply:
@haampie I have filed a internal ticket to get this sorted, it is being worked upon by @kiritigowda

Reply:
@haampie, our team have uploaded the patch for this issue - https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pull/524
Can you review that..
@kiritigowda